### PR TITLE
Infinite plasma pickaxe and plasma saw fix

### PR DESCRIPTION
**Pull request by DTraitor**
### Purpose

Fixes infinite plasma pickaxe and plasma saw. 
Right now you can turn on plasma pickaxe, remove it's battery and have infinite plasma pickaxe. This PR fixes it.

### CODE REVIEW

File: code/game/objects/items/weapons/tools/_tools.dm [hunk 1]
```diff
@@ -311,6 +311,8 @@
 //Cell reload
 /obj/item/weapon/tool/MouseDrop(over_object)
 	if((src.loc == usr) && istype(over_object, /obj/screen) && eject_item(cell, usr)) ///Formerly checked for /obj/screeninventory/hand. Not sure how that fits with bay's UI system
+		if (switched_on)
```
Comment (NanakoAC) [inline code/game/objects/items/weapons/tools/_tools.dm:314]: this doesn't seem the right place to do this

Anything with a power cost should be consuming power each tick, and turning off if it fails to do so. Find out why that isn't working

**Review by NanakoAC — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weapons/tools/_tools.dm (hunk 1 above)
Comment (DTraitor) [reply code/game/objects/items/weapons/tools/_tools.dm:314]: I've just checked few things and I've discovered that tools don't drain power passive. It drains power on use and when someone turns it on.

**Comment by DTraitor** [comment]
Tools will now turn off when you try to use them and there is no power left

File: code/game/objects/items/weapons/tools/_tools.dm (hunk 1 above)
Comment (NanakoAC) [reply code/game/objects/items/weapons/tools/_tools.dm:314]: passive power drain is definitely a thing, it must be broken., fixing that is the way to resolve this problem

**Comment by NanakoAC** [comment]
Dtraitor asked me to close this

**Pull request closed without merging**